src/Database/Type/DateTimeType.php

```diff
@@ -103,7 +103,7 @@ public function toDatabase($value, Driver $driver)
      */
     public function toPHP($value, Driver $driver)
     {
-        if ($value === null) {
+        if ($value === null || (int)$value === 0) {
```

Comment:
Why not check for allballs specifically? Int casting will fail on '0000-01-01 00:00:00' which is a 'valid' date.